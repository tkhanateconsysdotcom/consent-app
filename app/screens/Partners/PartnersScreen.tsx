import React from "react";
import { SafeAreaView, View } from "react-native";
import { useSelector } from "react-redux";
import { useNavigation } from "@react-navigation/native";

import { Text, Button } from "components";
import { consent } from "ducks";
import { t } from "utils";
import { navigate } from "navigation";

import styles from "./PartnersScreen.styles";
import { PartnersList } from "./components";
import { selectors } from "./ducks";

const PartnersScreen = () => {
  const monthlyAcceptanceConsent = useSelector(
    consent.selectors.getMonthlyAcceptanceConsent
  );
  const partners = useSelector(selectors.getPartners);
  const navigation = useNavigation();
  const navigator = navigate(navigation);

  return (
    <>
      <SafeAreaView style={styles.container}>
        <PartnersList
          monthlyAcceptanceConsent={monthlyAcceptanceConsent}
          partners={partners}
        />
      </SafeAreaView>
      <View style={styles.buttonView}>
        <Button.Primary
          fullWidth
          onPress={() => navigator.openAddPartner()}
          textType={"Secondary"}
        >
          <Text.Secondary numberOfLines={1} center white bold>
            {t("PARTNERS_SCREEN_ADD_PARTNER")}
          </Text.Secondary>
        </Button.Primary>
      </View>
    </>
  );
};

export default PartnersScreen;
