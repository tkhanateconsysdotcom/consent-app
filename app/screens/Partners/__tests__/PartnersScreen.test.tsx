import React from "react";
import { create } from "react-test-renderer";

import PartnersScreen from "../PartnersScreen";

it("PartnersScreen renders correctly", () => {
  const tree = create(<PartnersScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
