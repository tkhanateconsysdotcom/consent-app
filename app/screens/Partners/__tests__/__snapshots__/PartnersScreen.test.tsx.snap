// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnersScreen renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <PartnersList
      monthlyAcceptanceConsent={[MockFunction]}
      partners={[MockFunction]}
    />
  </RCTSafeAreaView>,
  <View
    style={
      Object {
        "bottom": 14,
        "flexDirection": "row",
        "left": 14,
        "position": "absolute",
        "right": 14,
      }
    }
  >
    <Button.Primary
      fullWidth={true}
      onPress={[Function]}
      textType="Secondary"
    >
      <Text.Secondary
        bold={true}
        center={true}
        numberOfLines={1}
        white={true}
      >
        Add partner
      </Text.Secondary>
    </Button.Primary>
  </View>,
]
`;
